src/resources/Catalogs/CatalogInterfaces.ts

```diff
@@ -426,3 +426,19 @@ export interface CatalogFieldStatsOptions {
      */
     forceRefresh?: boolean;
 }
+
+export interface CatalogMetadata {
+    /**
+     * Metadata seen on catalog documents with a sample of value.
+     */
+    metadataValuesByName: {
+        [name: string]: string[];
+    };
```

Comment:
[optional] can also be written as

```suggestion
    metadataValuesByName: Record<string, string>;
```